Automattic/wp-calypso
# Post Editor: reduxify save blockers in Post Edit Store

When uploading images, the post editor will not allow saving, previewing or publishing the post while the upload is in progress: the post content is in inconsistent, "transient" state.

This is implemented in Post Edit Store using the `blockSave` and `unblockSave` actions and the `PostEditStore.isSaveBlocked` method. This PR converts this code to Redux.

The first two commits are warmup drive-by changes:
- run the React ES6 class codemod on `PostEditor`. The class wasn't converted yet because it explicitly called `getInitialState` at a few places. After renaming the method to `getDefaultState` and using it to implement `getInitialState` the codemod ran successfully
- remove constructor from `EditorPublishButton`. It was only binding a `onClick` methods and there are more modern ways to achieve that

The third commit does the actual Reduxification.

**How to test:**
While in the post editor, drop an image into it. Verify that during the upload, in the editor top toolbar (aka "ground control"):
- "Save" button is not offered
- "Preview" and "Update" (or "Publish") buttons are disabled

After the upload is finished, verify that the above buttons are enabled again.

There are three ways how to upload an image that go through different code paths:
- drop the image in the visual (TinyMCE) view
- drop the image in the HTML view
- select "Add Media" in the editor toolbar and upload a file in the media library modal. Click "Insert" while the upload is still in progress. The modal will close and the editor will be in transient/blocked state until the upload finishes.
